src/components/BaseLayerSwitcher/BaseLayerSwitcher.js

```diff
@@ -0,0 +1,199 @@
+  className,
+  altText,
+  titles,
+}) {
+  const baseLayers = layers.filter(layer => layer.getIsBaseLayer());
+  const [switcherOpen, setSwitcherOpen] = useState(false);
+  const [isClosed, setIsClosed] = useState(true);
+  const [currentLayer, setCurrentLayer] = useState(getVisibleLayer(baseLayers));
+
+  /* Images are loaded from props if provided, fallback from layer */
+  const images = layerImages
+    ? Object.keys(layerImages).map(layerImage => layerImages[layerImage])
+    : baseLayers.map(layer => layer.previewImage);
+
+  const openClass = switcherOpen ? ' rs-open' : '';
+  const closedClass = isClosed ? ' rs-closed' : '';
+
+  const onLayerSelect = layer => {
+    if (!switcherOpen) {
+      setSwitcherOpen(true);
+      return;
+    }
+    setCurrentLayer(layer);
+    layer.setVisible(true);
+    setSwitcherOpen(false);
+  };
+
+  const nextImage = getNextImage(currentLayer, baseLayers, images);
+
+  useEffect(() => {
+    /* Ensure correct layer is active on app load */
+    if (currentLayer !== getVisibleLayer(baseLayers)) {
+      setCurrentLayer(getVisibleLayer(baseLayers));
+    }
+  }, [currentLayer]);
+
+  useEffect(() => {
+    /* Used for correct layer image render with animation */
+    let timeout;
+    if (!switcherOpen) {
+      timeout = setTimeout(() => {
+        setIsClosed(true);
+      }, 200);
+    } else {
+      setIsClosed(false);
+    }
+    return () => clearTimeout(timeout);
+  }, [switcherOpen]);
+
+  if (!baseLayers || baseLayers.length < 2) {
+    return null;
+  }
+
+  const toggleBtn = (
+    <div
+      className="rs-base-layer-switcher-button rs-close"
+      role="button"
+      onClick={() => setSwitcherOpen(false)}
+      onKeyPress={e => e.which === 13 && setSwitcherOpen(false)}
+      tabIndex="0"
+      aria-label={altText}
+      title={titles.closeSwitcher}
+    >
+      <FaChevronCircleLeft size={15} focusable={false} />
+    </div>
+  );
+
+  return (
+    <div className={`${className}${openClass}`}>
+      {!isClosed && toggleBtn}
+      {!isClosed ? (
+        baseLayers.map((layer, index) => {
+          const layerName = layer.getName();
+          const activeClass =
+            layerName === currentLayer.getName() ? ' rs-active' : '';
+          return (
+            <div
+              key={layer.key}
+              className={`rs-base-layer-switcher-button rs-layer${openClass}${activeClass}`}
+              role="button"
+              title={layerName}
+              aria-label={layerName}
+              onClick={() => onLayerSelect(layer)}
+              onKeyPress={e => {
+                if (e.which === 13) {
+                  onLayerSelect(layer);
+                }
+              }}
+              style={getImageStyle(images[index])}
+              tabIndex="0"
+            >
+              <div className="rs-base-layer-switcher-title">{layerName}</div>
+              {images[index] ? null : (
+                <span className="rs-alt-text">{altText}</span>
+              )}
```

Comment:
Is this a requirement this altText ?

Reply:
@mhaertwig did mention I should put in props